### ISSUE TITLE
[SITE] Update Flink docs to show supported versions

### ISSUE THREAD

**Issue by kbendick** [apache/iceberg#3881]
The Flink section of the website currently shows that only Flink version 1.11 is supported. https://iceberg.apache.org/#flink/

In the upcoming 0.13.0, Flink 1.12 / 1.13 / 1.14 will all be supported. I believe Iceberg 0.12.x also supports Flink 1.12, though possibly not Flink 1.13.

Now that we're explicitly tracking version compatibility in the source code, we should update the docs to reflect that.

**Comment by sanyud24** [comment]
May I work on this issue @kbendick ?

**Comment by kbendick** [comment]
So this is already done @sanyud24. I didn't realize we just need to deploy. But this is templated.

However, I'm going to keep this open or open a new ticket for needing to update the Flink docs to not be templated on the latest version anymore.

We are migrating to docs that will be by version, so this might not be an issue. I'll find out and tag you on a new ticket @sanyud24 👍

**Comment by kbendick** [comment]
I'm going to close this and take it offline.

Will reopen if the versioned docs don't already handle the templated version properly for the jar etc, but I think they do. We'le just waiting on the next version which should be out soon.

**Comment by sanyud24** [comment]
> 

Sure!! Thanks a lot!!
